### PR TITLE
Feature/endpoint skel

### PR DESCRIPTION
**Pull request by Jwilson1172**
# Description

This PR will add submission endpoint to the web app methods don't process anything yet but this sets the skeleton down for that processing to happen.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [X] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [X] Yes
- [ ] No
- [ ] Not necessary

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] There are no merge conflicts

### CODE REVIEW

File: project/app/api/submission.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        # XXX: (passing a hash then comparing that?)
+        # assert hashlib.sha256(data=file).hexdigest() == passed_hash
+        return value
+
+
+@router.post("/submission/text")
+async def submission_text(sub: Submission):
+    """Function that takes a file that is sent via FastAPI FileUpload interface
+    and uploads that file to an s3 bucket, then returns the link for the s3
+    bucket object and if the content has been flagged as inappropriate content.
```
Comment (schase15) [inline project/app/api/submission.py]: Has the file been transcribed to text at this point? If not, I think the text moderation step would need to come after it has been transcribed. If the transcription happens in this step can you mention it in the doc string? It was a little unclear to me

**Review by schase15 — CHANGES_REQUESTED** [review]
One comment on clarifying what's happening at the /submission/text endpoint. Other than that, structure and testing look great.

File: project/app/api/submission.py (hunk 1 above)
Comment (Jwilson1172) [reply project/app/api/submission.py]: ok I wasn't too sure how I was going to go about that, my plan was to have two async services, one that handles the s3 bucket upload, and one that handles the transcription and RDS entries. this method would call both services and wait for their responses then return that status to web with the link to the s3 bucket.

Comment (schase15) [reply project/app/api/submission.py]: Okay, then it is probably fine as-is for now. When you decide how you want to set it up and implement those pieces, just make sure to come back and clarify this.

**Review by schase15 — APPROVED** [review]

**Review by schase15 — APPROVED** [review]
Very clear description. Thank you

**Review by lorischl-otter — APPROVED** [review]
Looks great!

**Pull request merged**